src/mlpack/core/data/confusion_matrix.hpp

```diff
@@ -45,17 +45,35 @@ namespace data {
  * represents the predicted classes and column index represents the actual
  * class.
  *
- * @param predictors Vector of data points.
- * @param responses The measured data for each point.
+ * @param predictedClasses Vector of predicted classes for each data point.
+ * @param actualClasses Vector of actual classes for each data point.
  * @param output Matrix which is represented as confusion matrix.
  * @param numClasses Number of classes.
  */
 template<typename eT>
-void ConfusionMatrix(const arma::Row<size_t> predictors,
-                     const arma::Row<size_t> responses,
+void ConfusionMatrix(const arma::Row<size_t> predictedClasses,
+                     const arma::Row<size_t> actualClasses,
                      arma::Mat<eT>& output,
                      const size_t numClasses);
 
+/**
+ * Each value in this confusion matrix represent a percentage. The element at 
+ * (row, col)  represents the percentage of data points which were of category 'col' type 
+ * being predicted as 'row' type. 
+ * Note: here row index represents the predicted classes and column index represents 
+ * the actual class. 
+ * 
+ * @param predictedClasses Vector of predicted classes for each data point.
+ * @param actualClasses Vector of actual classes for each data point.
+ * @param output Matrix which is represented as confusion matrix.
+ * @param numClasses Number of classes.
+ */
+template<typename eT>
+void ConfusionMatrixPercentage(const arma::Row<size_t> predictedClasses,
```

Comment:
You can trivially produce this result by calling `ConfusionMatrix()` and then doing `output *= 100.0`; so I'm not sure what the advantage of adding this is.

Reply:
@rcurtin actually we cannot simply call `ConfusionMatrix()` and multiply each value in that Matrix by 100 and obtain the same result as the above proposed method does. `ConfusionMatrix()` only outputs the number of samples in each of the class. But in many cases we are not interested in knowing the exact number of samples of a class being predicted in some other class, rather we need a relative output, in form of percentages. 
A heat map will even be more useful to the end user, but since percentages also kinda do the same thing, this method will be useful for the end user.

Reply:
I see, thank you for the clarification!  I overlooked that and you are right.